src/configs/gemm-config.c

```diff
@@ -1745,6 +1747,28 @@ static void init_qp8_f32_qc4w_gemm_config(void) {
 #endif  // XNN_ARCH_ARM64 && XNN_ENABLE_KLEIDIAI
 }
 
+static void init_qp8_f32_qb4w_gemm_config(void) {
+#if XNN_ARCH_ARM64 && XNN_ENABLE_KLEIDIAI
+  const struct xnn_hardware_config* hardware_config =
+      xnn_init_hardware_config();
+  assert(hardware_config != NULL);
+  if (XNN_ENABLE_ARM_DOTPROD && hardware_config->use_arm_neon_dot) {
+#if XNN_ENABLE_ARM_DOTPROD
+    qp8_f32_qb4w_gemm_config.minmax.qp8gemm_bl[XNN_MR_TO_INDEX(1)] = xnn_init_hmp_qp8gemm_bl_ukernel(xnn_qp8_f32_qb4w_gemm_minmax_ukernel_1x8c16s2__aarch64_neondot);
```

Comment:
Just a note: From our experiments, we have found that the 8(mr)x4(kr) ukernel variant performs better than the 4(mr)x8(kr). If this is the case, we might need the 1x4 variant here to re-use the same RHS packed matrix. For now, we can keep this variant as we haven't integrated the GEMM variant kernel for the e2e test yet.

Reply:
hmm is the idea here that we fall back on the neon dot 1xkr ukernels when computing tiles where m < mr?

Reply:
This makes sense. When I create the config for i8mm i'll have the fallback (mr=1) be the 1x4 neondot kernel and i'll use the 8x4 neoni8mm kernel